overlord/snapstate/snapstate.go

```diff
@@ -2116,6 +2116,18 @@ func doUpdate(st *state.State, requested []string, updates []update, opts Option
 			continue
 		}
 
+		if up.SnapState.RefreshFailures != nil {
+			// Check if snap revision is known to fail and if the current refresh needs to be skipped.
+			if up.SnapState.RefreshFailures.ToRevision != up.Setup.Revision() {
+				// Snap has new target revision not known to fail, let's reset RefreshFailures
+				// and continue refresh normally.
+				up.SnapState.RefreshFailures = nil
```

Comment:
are we sure the SnapState  of the snap wasn't updated since it was put in up?

Reply:
Good point, I went through the call stack and SnapState is originated from:
- `storeUpdateGoal.toUpdate`
- `pathUpdateGoal.toUpdate`
- `refreshCandidate.SnapSetupForUpdate` (`autoRefreshPhase2` used in conditional auto-refreshes and continued auto-refreshes)

SnapState is not altered until it getting passed to `doUpdate`

Reply:
thanks for checking this